pennylane/qnodes/jacobian.py

```diff
@@ -23,25 +23,54 @@
 
 from .base import BaseQNode, QuantumFunctionError
 
+DEFAULT_STEP_SIZE = 0.3
+DEFAULT_STEP_SIZE_ANALYTIC = 1e-7
+
+
+def default_step_size(analytic):
+    """Returns the step size based on the boolean.
+
+    Args:
+        analytic (bool): determines if the analytic step size is to be returned
+
+    Returns:
+        float: the step size requested
+    """
+    return DEFAULT_STEP_SIZE_ANALYTIC if analytic else DEFAULT_STEP_SIZE
+
 
 class JacobianQNode(BaseQNode):
     """Quantum node that can be differentiated with respect to its positional parameters.
     """
 
-    def __init__(self, func, device, mutable=True, properties=None):
+    def __init__(self, func, device, mutable=True, h=None, properties=None):
         super().__init__(func, device, mutable=mutable, properties=properties)
 
         self.par_to_grad_method = None
         """dict[int, str]: map from flattened quantum function positional parameter index
         to the gradient method to be used with that parameter"""
 
+        self._h = h or default_step_size(device.analytic)
```

Comment:
Nitpick: I find it more readable to have this logic be all in the same place:
```python
default_step_size = DEFAULT_STEP_SIZE_ANALYTIC if analytic else DEFAULT_STEP_SIZE
self._h = h or default_step_size 
```
I'm not sure that this logic is complex enough to warrant the overhead of a separate function.

(At some point it may be the case that the logic for finding the default _is_ more complicated, in which case we should move it to its own function, but I feel it is premature to do it now).

Reply:
Sounds good to me :)

Reply:
(The `device.analytic` attribute would be used here for the branching logic.)